x-pack/plugins/observability/public/embeddable/slo/alerts/components/slo_alerts_table.tsx

```diff
@@ -33,7 +38,7 @@ export const getSloInstanceFilter = (sloId: string, sloInstanceId: string) => {
             'slo.id': sloId,
           },
         },
-        ...(sloInstanceId !== ALL_VALUE
+        ...(sloInstanceId !== ALL_VALUE && !showAllGroupByInstances
```

Comment:
It's clear what's happening. In my example it will return an empty array, but still it will return all SLOs with that slo.id. Sorry I missed the term slo.id above.